kotlinx-coroutines-debug/src/junit4/CoroutinesTimeoutStatement.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.debug.junit4
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.debug.*
+import org.junit.runner.*
+import org.junit.runners.model.*
+import java.util.concurrent.*
+
+internal class CoroutinesTimeoutStatement(
+    private val testStatement: Statement, private val testDescription: Description,
+    private val testTimeoutMs: Long,
+    private val cancelOnTimeout: Boolean = false
+) : Statement() {
+
+    private val testExecutor = Executors.newSingleThreadExecutor {
+        Thread(it).apply {
+            name = "Timeout test executor"
+            isDaemon = true
+        }
+    }
+
+    // Thread to dump stack from, captured by testExecutor
+    private lateinit var testThread: Thread
+
+    override fun evaluate() {
+        DebugProbes.install() // Fail-fast if probes are unavailable
+        val latch = CountDownLatch(1)
+        val testFuture = CompletableFuture.runAsync(Runnable {
+            testThread = Thread.currentThread()
+            latch.countDown()
+            testStatement.evaluate()
+        }, testExecutor)
+
+        latch.await() // Await until test is started
+        try {
+            testFuture.get(testTimeoutMs, TimeUnit.MILLISECONDS)
+            return
+        } catch (e: TimeoutException) {
+            handleTimeout(testDescription)
+        } catch (e: ExecutionException) {
+            throw e.cause ?: e
+        } finally {
+            DebugProbes.uninstall()
+            testExecutor.shutdown()
+        }
+    }
+
+    private fun handleTimeout(description: Description) {
+        val units =
+            if (testTimeoutMs % 1000L == 0L)
+                "${testTimeoutMs / 1000} seconds"
```

Comment:
`1000L` on one line and `1000` (without `L`) on the other looks like inconsistent style. Consider using just `1000` on both lines.